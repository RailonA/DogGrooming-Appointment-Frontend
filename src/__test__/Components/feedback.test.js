import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('Feedback', () => {
  const mockStore = configureStore([]);
  const store = mockStore({});
  store.dispatch = jest.fn();
  const type = 'error';
  const feedbackMessage = 'Error text';

  it('should match the snapshot', () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <div feedback={feedbackMessage} type={type}>
          <div>
            <div
              className="modal-header"
            >
              <div
                className="modal-title h4"
              >
                Error
              </div>
            </div>
            <div
              className="modal-body"
            >
              <p>
                Error text
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-secondary"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
            </div>
          </div>
        </div>
      </Provider>,
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
