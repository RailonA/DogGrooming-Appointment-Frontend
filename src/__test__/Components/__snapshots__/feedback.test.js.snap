// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match the snapshot 1`] = `
<div
  feedback="Error text"
  type="error"
>
  <div>
    <div
      className="modal-header"
    >
      <div
        className="modal-title h4"
      >
        Error
      </div>
    </div>
    <div
      className="modal-body"
    >
      <p>
        Error text
      </p>
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={
          Array [
            [Function],
          ]
        }
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
